perun-web-gui/src/main/java/cz/metacentrum/perun/webgui/tabs/registrartabs/EditFormItemTabItem.java

```diff
@@ -531,6 +531,15 @@ public void onFinished(JavaScriptObject jso) {
 					// sort
 					list = new TableSorter<AttributeDefinition>().sortByFriendlyName(list);
 					for (AttributeDefinition def : list) {
+
+						// allow ArrayList and LinkedHashMap attributes only to designated widgets, unless it was already selected
+						boolean isMapOrListWidget = item.getType().equalsIgnoreCase("MAP_INPUT_WIDGET") || item.getType().equalsIgnoreCase("LIST_INPUT_WIDGET");
+						boolean isMapOrListAttribute = def.getType().contains("ArrayList") || def.getType().contains("LinkedHashMap");
+						boolean alreadySelected = item.getPerunSourceAttribute().contains(def.getFriendlyName()) || item.getPerunDestinationAttribute().contains(def.getFriendlyName());
```

Comment:
To prevent null pointer when source/dest attribute is not yet set on the form item. Also we should compare whole attr name, not just friendly name.
```suggestion
						boolean alreadySelected = def.getName().equalsIgnoreCase(item.getPerunSourceAttribute()) || def.getName().equalsIgnoreCase(item.getPerunDestinationAttribute());
```